File: ios/CrisisMap/Models/SourceTransparencyContent.swift
import Foundation

struct SourceTransparencyContent {
    struct Source: Identifiable, Equatable {
        let name: String
        let detailLocalizationKey: String?

        var id: String { name }
    }

    struct Limitation: Identifiable, Equatable {
        let localizationKey: String

        var id: String { localizationKey }
    }

    static let newsSources: [Source] = [
        Source(name: "Reuters", detailLocalizationKey: nil),
        Source(name: "AP News", detailLocalizationKey: nil),
        Source(name: "BBC News", detailLocalizationKey: nil),
        Source(name: "NHK World", detailLocalizationKey: nil),
        Source(name: "Al Jazeera", detailLocalizationKey: nil),
        Source(name: "DW", detailLocalizationKey: nil),
        Source(name: "The Guardian", detailLocalizationKey: nil),
        Source(name: "NPR World", detailLocalizationKey: nil),
        Source(name: "France 24", detailLocalizationKey: nil),
        Source(name: "UN News", detailLocalizationKey: nil),
        Source(name: "GDELT", detailLocalizationKey: "sourceTransparency.news.gdelt"),
        Source(name: "X/Grok", detailLocalizationKey: "sourceTransparency.news.xGrok")
    ]

    static let researchSources: [Source] = [
        Source(name: "Brookings", detailLocalizationKey: nil),
        Source(name: "CATO", detailLocalizationKey: nil),
        Source(name: "CFR", detailLocalizationKey: nil),
        Source(name: "CSIS", detailLocalizationKey: nil),
        Source(name: "Chatham House", detailLocalizationKey: nil),
        Source(name: "Foreign Affairs", detailLocalizationKey: nil),
        Source(name: "Heritage", detailLocalizationKey: nil),
        Source(name: "IISS", detailLocalizationKey: nil),
        Source(name: "INSS", detailLocalizationKey: nil),
        Source(name: "Mitchell", detailLocalizationKey: nil),
        Source(name: "RAND", detailLocalizationKey: nil),
        Source(name: "USNI", detailLocalizationKey: nil)
    ]

    static let limitations: [Limitation] = [
        Limitation(localizationKey: "sourceTransparency.limit.coverage"),
        Limitation(localizationKey: "sourceTransparency.limit.availability"),
        Limitation(localizationKey: "sourceTransparency.limit.originals")
    ]
}

File: android/app/src/main/java/com/crisismap/app/ui/shared/SourceTransparencyContent.kt
package com.crisismap.app.ui.shared

import androidx.annotation.StringRes
import com.crisismap.app.R

data class TransparencySource(
    val name: String,
    @StringRes val detailRes: Int? = null
)

data class TransparencyLimitation(
    @StringRes val textRes: Int
)

object SourceTransparencyContent {
    val newsSources = listOf(
        TransparencySource(name = "Reuters"),
        TransparencySource(name = "AP News"),
        TransparencySource(name = "BBC News"),
        TransparencySource(name = "NHK World"),
        TransparencySource(name = "Al Jazeera"),
        TransparencySource(name = "DW"),
        TransparencySource(name = "The Guardian"),
        TransparencySource(name = "NPR World"),
        TransparencySource(name = "France 24"),
        TransparencySource(name = "UN News"),
        TransparencySource(name = "GDELT", detailRes = R.string.source_transparency_news_gdelt),
        TransparencySource(name = "X/Grok", detailRes = R.string.source_transparency_news_x_grok)
    )

    val researchSources = listOf(
        TransparencySource(name = "Brookings"),
        TransparencySource(name = "CATO"),
        TransparencySource(name = "CFR"),
        TransparencySource(name = "CSIS"),
        TransparencySource(name = "Chatham House"),
        TransparencySource(name = "Foreign Affairs"),
        TransparencySource(name = "Heritage"),
        TransparencySource(name = "IISS"),
        TransparencySource(name = "INSS"),
        TransparencySource(name = "Mitchell"),
        TransparencySource(name = "RAND"),
        TransparencySource(name = "USNI")
    )

    val limitations = listOf(
        TransparencyLimitation(textRes = R.string.source_transparency_limit_coverage),
        TransparencyLimitation(textRes = R.string.source_transparency_limit_availability),
        TransparencyLimitation(textRes = R.string.source_transparency_limit_originals)
    )
}
